src/com/github/kornilova203/matlab/completion/MatlabLibraryFunctionCompletionContributor.kt

```diff
@@ -0,0 +1,100 @@
+package com.github.kornilova203.matlab.completion
+
+import com.github.kornilova203.matlab.completion.MatlabKeywordCompletionContributor.Companion.M
+import com.github.kornilova203.matlab.psi.MatlabQualifiedExpr
+import com.github.kornilova203.matlab.psi.MatlabRefExpr
+import com.github.kornilova203.matlab.psi.MatlabTypes
+import com.intellij.codeInsight.completion.*
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.PlatformPatterns.psiElement
+import com.intellij.patterns.StandardPatterns.*
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import java.io.BufferedReader
+import javax.xml.parsers.DocumentBuilderFactory
+import kotlin.collections.HashSet
+
+class MatlabLibraryFunctionCompletionContributor : CompletionContributor() {
+    companion object {
+        private var functions = HashSet<LookupElement>()
+        private var packages = HashMap<String, MutableSet<LookupElement>>()
+
+        val KEYWORDS = hashSetOf("function", "end", "if", "else", "elseif", "while", "switch", "case", "otherwise", "for", "classdef", "try", "catch", "global")
+
+        private val IDENT = psiElement(MatlabTypes.IDENTIFIER).withParent(MatlabRefExpr::class.java)
+        private val IN_QUALIFIED_EXPR = and(M, IDENT.withSuperParent(2, MatlabQualifiedExpr::class.java))
+        private val AFTER_NUMBER_LITERAL = psiElement().afterLeafSkipping(
+                alwaysFalse<PsiElement>(),
+                or(psiElement(MatlabTypes.INTEGER), psiElement(MatlabTypes.FLOAT))
+        )
+    }
+
+    init {
+        getDocs()
+        extend(CompletionType.BASIC,
+                M,
+                object : CompletionProvider<CompletionParameters>() {
+                    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+                        val element = parameters.position
+                        if (AFTER_NUMBER_LITERAL.accepts(element)) {
+                            result.stopHere()
+                            return
+                        }
+                        if (IN_QUALIFIED_EXPR.accepts(element)) {
+                            val valueBeforePoint = element.parent.prevSibling.prevSibling.lastChild.text
+                            if (packages.containsKey(valueBeforePoint)) {
+                                result.addAllElements(packages[valueBeforePoint]!!)
+                            }
+                        } else {
+                            result.addAllElements(functions)
+                        }
+                    }
+                })
+    }
+
+    private fun getDocs() {
+        try {
+            val doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse("https://www.mathworks.com/help/matlab/referencelist_function_cat.xml")
```

Comment:
`getDocs` is invoked on ui thread so I'm afraid this line may cause ui freeze.
I think local file with functions list is good enough for now and there is no need to fetch functions from internet.